python/src/uagents/agent.py

```diff
@@ -761,10 +761,11 @@ def include(self, protocol: Protocol, publish_manifest: Optional[bool] = False):
         self._interval_messages.update(protocol.interval_messages)
 
         for schema_digest in protocol.models:
-            if schema_digest in self._models:
-                raise RuntimeError("Unable to register duplicate model")
-            if schema_digest in self._signed_message_handlers:
-                raise RuntimeError("Unable to register duplicate message handler")
+            if not hasattr(protocol, "rules"):
```

Comment:
I still don't see the need for this. The overwriting of the default behavior by the decorator happens before `include` is called, so this is not triggered. Once we are here, there will be only one model entry with the function registered by the latest decorator (or latest call of `set_default_behaviour`). It runs perfectly for me without this line.

There is, however, one issue that is also not caught by this: If we overwrite the transition with a decorator specifying another `Model` than what was used in `set_default_behaviour`, (e.g., `@chitchat_dialogue.on_init_session(RejectChitChatDialogue)` the agent will crash, as this is a yet undefined state. We would either need to enforce that if a default handler has been set, overwriting is only possible using the same `Model` or make sure to completely remove the earlier  registration.

Reply:
You are right, this must have been an issue that was already handled / fixed by the other commits. I definitely remember having to add this line to make it work but it doesn't seem to be necessary any more.

As for the other point, it makes sense to restrict overwrites of edges' handlers to the same message Models as both participants need to support the same models (due to the digests in the manifest) to properly communicate. Ultimately this comes down to how and what we actually put into the manifest going forward.

Reply:
With the current implementation of manifests this would eliminate the possibility of defining a general pattern that can be instantiated with different models, as soon as a default behavior is defined. Conceptually, it would be nicer to still keep patterns general and provide default behavior with concrete dialogues. However, I'm not sure if the current architecture really supports this, since the distinction between the logical `Pattern` and `Dialogue` is only based on whether a `Dialogue` is instantiated or not...

I guess this the long way of saying, I agree it should just be restricted in this version and needs to be reevaluated in the bigger picture.

Reply:
If the models are used as a general purpose vehicle this can still work. I'm thinking about a similar approach to what Attila has done with the Metadata where the contents of the message can still be overloaded to contain custom information while retaining the connection of message<->interaction

Reply:
Going to close this conversation since we take it to a separate meeting.